File: src/m/parameterization/sethydrostaticmask.py
import numpy as np


def setmask(md):
    """
    SETHYDROSTATICMASK - establish ocean_levelset field

   Determines grounded and floating ice position based on
   md.geometry.bed and md.geometry.thickness

   Usage:
      md = sethydrostaticmask(md)

   Examples:
      md = sethydrostaticmask(md)
   """

    if np.size(md.geometry.bed, axis=0) != md.mesh.numberofvertices or np.size(md.geometry.base, axis=0) != md.mesh.numberofvertices or np.size(md.geometry.thickness, axis=0) != md.mesh.numberofvertices:
        raise IOError("hydrostaticmask error message: fields in md.geometry do not have the right size.")

    # grounded ice level set
    md.mask.ocean_levelset = md.geometry.thickness + md.geometry.bed * md.materials.rho_water / md.materials.rho_ice

    #Check consistency of geometry
    if any(md.geometry.base[np.nonzero(md.mask.ocean_levelset > 0.)] != md.geometry.bed[np.nonzero(md.mask.ocean_levelset > 0.)]):
        print("WARNING: md.geometry.bed and md.geometry.base not equal on grounded ice")

    if any(md.geometry.base[np.nonzero(md.mask.ocean_levelset <= 0.)] < md.geometry.bed[np.nonzero(md.mask.ocean_levelset <= 0.)]):
        print("WARNING: md.geometry.base < md.geometry.bed on floating ice")

    return md

File: src/m/parameterization/sethydrostaticmask.m
function ocean_levelset = sethydrostaticmask(md)
%SETHYDROSTATICMASK - establish ocean_levelset field
%
%   Determines grounded and floating ice position based on 
%   md.geometry.bed and md.geometry.thickness
%
%   Usage:
%      ocean_levelset = sethydrostaticmask(md)
%
%   Examples:
%      md.mask.ocean_levelset = sethydrostaticmask(md);

%temporary warning for people that already use this function
disp('Setting hydrostatic mask: WARNING - now returning ocean_levelset (not md)');
disp('   -- you can blame Mathieu');

if(length(md.geometry.bed)~=md.mesh.numberofvertices | length(md.geometry.thickness)~=md.mesh.numberofvertices | length(md.geometry.base)~=md.mesh.numberofvertices),
		error('hydrostaticmask error message: fields in md.geometry do not have the right size.');
end

%ocean level set based on height above floatation
ocean_levelset=md.geometry.thickness+md.geometry.bed*md.materials.rho_water/md.materials.rho_ice;

%Check consistency of geometry
pos=find(ocean_levelset>0);
if(any(md.geometry.base(pos)~=md.geometry.bed(pos))),
	disp('WARNING: md.geometry.bed and md.geometry.base not equal on grounded ice');
end

pos=find(ocean_levelset<=0);
if(any(md.geometry.base(pos)<md.geometry.bed(pos))),
	disp('WARNING: md.geometry.base < md.geometry.bed on floating ice');
end
